openstf/model/split_energy.py

```diff
@@ -42,23 +44,65 @@ def split_energy(pid):
     error = components[["load", "Inschatting"]].diff(axis=1).iloc[:, 1]
     mae = error.abs().mean()
     coefdict.update({"MAE": mae})
+    coefsdf = convert_coefdict_to_coefsdf(pj, input_split_function, coefdict)
 
-    # Get average coefs of previous runs
+    # Get average coefs of previous runs and check if new coefs are valid
     mean_coefs = db.get_energy_split_coefs(pj, mean=True)
-    # Loop over keys and check if the difference with the average value is not more than 100%
-    # In case the difference is more tha 100% of the average set KPI coefs as expected to False
+    if are_new_coefs_valid(coefdict, mean_coefs) is False:
+        # If coefs not valid, do not update the coefs in the db and send teams
+        # message that something strange is happening
+        monitoring.post_teams_alert(
+            "New splitting coefficients for pid {} deviate strongly from previously stored coefficients.".format(
+                pj["id"],
+            ),
+            coefsdf=coefsdf,
+        )
+        # Use the last known coefficients for further processing
+        last_coefdict = db.get_energy_split_coefs(pj)
+        last_coefsdf = convert_coefdict_to_coefsdf(
+            pj, input_split_function, last_coefdict
+        )
+        return last_coefsdf
+
+    # Save Results
+    db.write_energy_splitting_coefficients(coefsdf, if_exists="append")
```

Comment:
An explicit 'else: ' here is more clear (although it functionally doesn't matter)

Reply:
Ok